tools/autograd/derivatives.yaml

```diff
@@ -1952,7 +1952,12 @@
 
 - name: log_sigmoid_forward(Tensor self) -> (Tensor output, Tensor buffer)
   self: log_sigmoid_backward(grad, self, buffer)
-  output: auto_element_wise
+  # HACK: We add a `view_as` to auto_element_wise, so that we no longer have to
+  # use the jvp decomposition trick in functorch, but we also don't want
+  # test_vmapjvpall_nn_functional_logsigmoid_cuda_float32 to regress (passes on cpu!)
+  # We should either figure out what is going on with vmap or perhaps fwd AD could
+  # be more tolerant about 0-dim vs 1-dim tensors
+  output: log_sigmoid_backward(self_t.conj(), self_p, buffer).conj().view_as(self_p)
```

Comment:
Forward AD was complaining about trying to set a scalar foward grad to a 1-D tensor. The reason we have this now is previously we called into the c++ composite op decomposition with `OP_DECOMPOSE`, but now we use the forward AD formula directly.